client/verta/verta/_internal_utils/_utils.py

```diff
@@ -584,11 +589,11 @@ def raise_for_http_error(response):
                 cause = "Server"
             else:  # should be impossible here, but sure okay
                 cause = "Unexpected"
-            message = "{} {} Error: {} for url: {}".format(
-                response.status_code, cause, reason, response.url
-            )
-            message += time_str  # attach time to error message
-            six.raise_from(requests.HTTPError(message, response=response), None)
+            message = f"{response.status_code} {cause} Error: {reason} " \
+                      f"for url: {response.url}{time_str}"
+            e.args = (message,)
+        traceback.print_exc(file=None, limit=traceback_limit)
```

Comment:
Unfortunately, this doesn't seem to do what we'd like 😕

I was hoping that it would cut down on the stack trace of the exception itself. Instead, what it seems to do is print the traceback to `stderr` and leave the exception unchanged.

Reply:
This means that `HTTPError`s throughout the client now get effectively duplicated output

![Screen Shot 2022-12-02 at 3 27 52 PM](https://user-images.githubusercontent.com/96442646/205406816-ccf32715-7a01-4dde-a5ea-563b6b4bd9df.png)

Reply:
And model prediction errors still output their massive stack trace (and it's not visually clear to a user what part of this—if any—came from their model, vs being some other error with the client/platform)
![Screen Shot 2022-12-02 at 3 18 42 PM](https://user-images.githubusercontent.com/96442646/205406950-c97dd70d-e1ff-4a4c-aaf3-7b3a2b76df20.png)

Reply:
Perhaps at this point we ought to keep the original logic

```python
            if response.status_code in (
                400,
                502,
            ):  # possibly error from the model back end
                try:
                    data = _utils.body_to_json(response)
                except ValueError:  # not JSON response; 502 not from model back end
                    pass
                else:  # from model back end; contains message (maybe)
                    # try to directly print message, otherwise line breaks appear as '\n'
                    msg = data.get("message") or json.dumps(data)
                    raise RuntimeError(
                        "deployed model encountered an error: {}".format(msg)
                    )
```

maybe in a standalone function so that it's not entangled with other code, and would be easier to tear out in the near future.